test/playwright/utils/navigation.ts

```diff
@@ -89,8 +116,135 @@ export const currentContentType = async (page: Page) => {
+  const dir = options.dir || 'ltr'
+  const mode = options.mode ?? 'SSR'
+  const query = options.query ? `&${options.query}` : ''
+
+  if (mode === 'SSR') {
+    await page.goto(
+      pathWithDir(`search/${searchTypePath(searchType)}?q=${term}${query}`, dir)
+    )
+    await dismissTranslationBanner(page)
+  } else {
+    await page.goto(pathWithDir('/', dir))
+    await dismissTranslationBanner(page)
+    // Select the search type
+    if (searchType !== 'all') {
+      await page.click('[aria-label="All content"]')
+      await page.click(
+        `button[role="radio"]:has-text("${searchTypeNames[searchType]}")`
+      )
+    }
+    // Type search term
+    const searchInput = page.locator('main input[type="search"]')
+    await searchInput.type(term)
+    // Click search button
+    // Wait for navigation
+    await Promise.all([
+      page.click('[aria-label="Search"]'),
+      page.waitForNavigation(),
+    ])
+    await scrollDownAndUp(page)
+  }
+}
+
+/**
+ * Fills the search input in the page header, clicks on submit
+ * and waits for navigation.
+ */
+export const searchFromHeader = async (page: Page, term: string) => {
+  await page.fill('id=search-bar', term)
+  await Promise.all([
+    page.waitForNavigation(),
+    page.locator('button[type="submit"]').click(),
+  ])
+}
+
+/**
+ * Click on the first <mediaType> result: a link that contains
+ * /<mediaType>/ in its URL. We cannot use the 'startsWith' `^` matcher
+ * because localized routes start with the locale prefix (e.g. /ar/image/).
+ * Scroll down and up to load all lazy-loaded content.
+ */
+export const openFirstResult = async (page: Page, mediaType: MediaType) => {
+  await page.locator(`a[href*="/${mediaType}/"]`).first().click()
+  await scrollDownAndUp(page)
+}
+
+/**
+ * Click on the first <mediaType> related result: a link that contains
+ * /<mediaType>/ in its URL in the 'aside' element for related media.
+ * We cannot use the 'startsWith' `^` matcher because localized routes
+ * start with the locale prefix (e.g. /ar/image/).
+ * Scroll down and up to load all lazy-loaded content.
+ */
+export const openFirstRelatedResult = async (
+  page: Page,
+  mediaType: MediaType
+) => {
+  await page.locator(`aside a[href*="/${mediaType}/"]`).first().click()
+  await scrollDownAndUp(page)
+}
+
+export const scrollToBottom = async (page: Page) => {
+  await page.evaluate(() => {
+    window.scrollTo(0, document.body.scrollHeight)
+  })
+  await sleep(300)
+}
+
+export const scrollToTop = async (page: Page) => {
+  await page.evaluate(() => {
+    window.scrollTo(0, 0)
+  })
+  await sleep(200)
+}
+
+/**
+ * Used to load all lazy-loaded images in the page.
+ */
+export const scrollDownAndUp = async (page: Page) => {
+  await scrollToBottom(page)
+  await scrollToTop(page)
+}
+
+export const renderDirs = ['ltr', 'rtl'] as const
+
+export const renderModes = [
+  ['SSR', 'ltr'],
+  ['SSR', 'rtl'],
+  ['CSR', 'ltr'],
+  ['CSR', 'rtl'],
+] as const
```

Comment:
Kind of a nit but maybe not because render is a verb and a noun so there is some potential for confusion here: what is render meant to indicate here? Could `renderDirs` just be `languageDirs` or `pageDirections`? I'd prefer `directions` in this case (and in the `pageWithDir` function name) because `dir` is also easily confused with "directory" (in fact it took me a second to understand what `dir` meant in the `pageWithDir` function based on the signature and name of the function).

`renderModes` might be clearer as `renderingContexts`.

What do you think?

Reply:
I am always grateful for naming suggestions! I wonder if it should be `languageDirections` to be very clear?
And `renderingContexts` is perfect!

Reply:
`languageDirections` is perfect!